.github/workflows/test-all.yml

```diff
@@ -0,0 +1,190 @@
+name: "Test all packages"
+
+on:
+  workflow_dispatch:
+    inputs:
+      pkgarchiveurl:
+        description: 'URL of the package archive to test'
+        # TODO: instead / alternatively, specify a commit in this repo and then
+        # assemble a tarball from that?
+        required: true
+        type: string
+        default: https://github.com/gap-system/PackageDistro/releases/download/latest/packages.tar.gz
+  workflow_call:
+    inputs:
+      pkgarchiveurl:
+        required: true
+        type: string
+
+jobs:
+  build:
+    name: "Build GAP and the packages"
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.get-names.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: "Setup _archives symlink"
+        run: mkdir -p ~/_archives && ln -s ~/_archives
+
+      - name: "Cache archives"
+        uses: actions/cache@v2
+        with:
+          path: ~/_archives
+          key: archives-${{ hashFiles('**/meta.json') }}
+          restore-keys: |
+            archives-
+
+      - name: "Install package distribution tools"
+        run: |
+          git clone https://github.com/gap-system/PackageDistroTools _tools
+          pip install -r _tools/requirements.txt
+
+      # TOOD: dependencies should come from a container
+      - name: "Install binary package dependencies"
+        run: |
+          deps=$(_tools/gather_dependencies.py */meta.json)
+          if [[ -n "$deps" ]]; then
+            echo "Installing required binary depedencies: $deps"
+            sudo apt-get install $deps
+          else
+            echo "No required binary depedencies to be installed"
+          fi
+
+      - name: "Download packages"
+        run: |
+          #curl -L -o $HOME/packages.tar.gz ${{ github.event.inputs.pkgarchiveurl || inputs.pkgarchiveurl }}
+          #ls -l $HOME/packages.tar.gz
+          #shasum -a 256 $HOME/packages.tar.gz
+          ## TODO: also download and test checksum???
+          _tools/download_packages.py */meta.json
```

Comment:
Not sure how to best "download packages":
1. using `download_packages.py` has the advantage that we really test the packages in the commit this action runs on
2. downloading a `packages.tar.gz` from a given ensures we really test the tarball we later offer to users; but it requires a download URL for such a package

In practice, I envision that this is run from either ...
1. a PR workflow (to test whether a PR is ready to be merged),
2. upon a `push` to the `main` branch (to test a `packages.tar.gz` we just assembled),
3. as part of a scheduled cron job, to test for regressions introduced in GAP `master`

For the first scenario, it makes sense to use `download_packages.py`; for the other two scenarios, I'd say the second approach makes more sense. So perhaps both should be supported, by turning the `pkgarchiveurl` into an optional argument?